### ISSUE TITLE
Limit fireworks

### ISSUE THREAD

**Issue by JeromSar** [TotalFreedom/TotalFreedomMod#899]
Mass amounts of fireworks can lag clients. Rate limiting fireworks may be a good idea.

**Comment by ChargedCreeper** [comment]
I agree. I've experienced crashes from people rapidly spamming fireworks in the past.

**Comment by Hockeyfan360** [comment]
Keep in mind that you would have to find a way to prevent dispensers from shooting fireworks at rapid rates.

**Comment by Wild1145** [comment]
@Hockeyfan360 Or we just disable dispensers being able to shoot them full stop?

**Comment by Hockeyfan360** [comment]
@Wild1145 That'd be a big breach of freedom.

I think that admins should be alerted that someone is spamming them rapidly. If they have bad intentions, they can be banned. If they are putting on a firework show or similar, they shouldn't be punished/prevented.

**Comment by Wild1145** [comment]
@Hockeyfan360 And in the time it takes for an admin (If there is one online and not pissing around) to actually respond to the issue the server could already be crashing. While I do not wish to remove the freedom from the players, a line has to be drawn somewhere. Limiting the amount of entities that a dispenser can launch will likely prove to be difficult at best. I personally can see no practical way of implementing this without putting global entity limits in place, and this would still not solve the root cause.

**Comment by JeromSar** [comment]
How about limiting the amount of firework entities that may exist in a given chunk. Any new spawned firework entities could be blocked and removed.

**Comment by Hockeyfan360** [comment]
@JeromSar I think that that would be good.
